File: tutorials/csharp-tutorial/src/CSharpTutorialDemo/CSharpTutorialDemo/Arrays.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharpTutorialDemo
{
    public class Arrays
    {
        public static void CreateAndThenInitialize()
        {
            int[] myArray = new int[6];

            myArray[0] = 1;
            myArray[1] = 2;
            myArray[2] = 3;
            myArray[3] = 4;
            myArray[4] = 5;
            myArray[5] = 6;

            for (int i = 0; i < 6; i++)
                Console.WriteLine(myArray[i]);
        }

        public static void InitializeWithDefaultValues()
        {
            int[] myArray = { 1, 2, 3, 4, 5, 6 };
            string[] daysOfWeek = { "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday" };

            for (int i = 0; i < 6; i++)
                Console.WriteLine(myArray[i]);

            for (int i = 0; i < 7; i++)
                Console.WriteLine(daysOfWeek[i]);
        }

        public static void AccessArray()
        {
            int[] myArray = { 1, 2, 3, 4, 5, 6 };
            
            for (int i = 0; i < 6; i++)
                Console.WriteLine(myArray[i]);

            int index = 3;

            myArray[index] = 100;

            for (int i = 0; i < 6; i++)
                Console.WriteLine(myArray[i]);
        }
    }
}

File: tutorials/vb-net/src/VBNetTutorialDemo/VBNetTutorialDemo/Arrays.vb
Module Arrays
    Public Sub Example1()
        Dim myArray1 As Integer()
        Dim myArray2 As Integer() = New Integer(5) {}
        Dim myArray3(5) As Double
        Dim myArray4 As Integer() = {1, 2, 3, 4, 5, 6}
        Dim daysOfWeek As String() = {"Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"}
        myArray4(3) = 100
        Console.WriteLine(myArray4(3))
    End Sub

    Public Sub Example2()
        Dim myArray4 As Integer() = {1, 2, 3, 4, 5, 6}

        ' Assign a new array size and retain the current values.
        ReDim Preserve myArray4(20)

        ' Assign a new array size and retain only the first five values.
        ReDim Preserve myArray4(4)

        ' Assign a new array size and discard all current element values.
        ReDim myArray4(15)
    End Sub

    Public Sub Example3()
        ' Create a 10-element integer array.
        Dim numbers(9) As Integer
        Dim value As Integer = 2

        ' Write values to it.
        For ctr As Integer = 0 To 9
            numbers(ctr) = value
            value *= 2
        Next

        ' Read and sum the array values.  
        Dim sum As Integer
        For ctr As Integer = 0 To 9
            sum += numbers(ctr)
        Next
        Console.WriteLine($"The sum of the values is {sum:N0}")
    End Sub
End Module
